src/Icicle/Source/PrettyAnnot.hs

```diff
@@ -1,74 +1,133 @@
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE PatternGuards     #-}
 module Icicle.Source.PrettyAnnot (
     PrettyAnnot(..)
   ) where
 
+import                  Icicle.Source.Type
 import                  Icicle.Source.Query
 import                  Icicle.Internal.Pretty
 
 import                  P
 
-import                  Data.List (intersperse)
-
 newtype PrettyAnnot q
  = PrettyAnnot
  { getPrettyAnnot :: q }
 
-instance (Pretty a, Pretty n) => Pretty (PrettyAnnot (Exp a n)) where
+instance (Pretty a, Pretty n) => Pretty (PrettyAnnot (Exp (Annot a n) n)) where
```

Comment:
oh cool, this is good.
do you suppose there's any way to add these annotations to the ``Pretty`` instance in ``Icicle.Source.Query.Query``, and selectively filter them out or not print them afterwards?
I think adding ``Pretty a`` to the instance there is reasonable.